### PR TITLE
190 dataprovider enchantments

### PR DESCRIPTION
**Pull request by oliverde8**
**Don't merge this for now**

### CODE REVIEW

**Comment by oliverde8** [comment]
This can be merged now might introduce new issues but globally it's better and adds more flexibility.

File: src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/WaypointDataListenerInterface.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace eXpansion\Framework\GameTrackmania\DataProviders\Listener;
+
+/**
+ * Class RaceDataListenerInterface
+ *
+ * @package eXpansion\Framework\GameTrackmania\DataProviders\Listener;
+ * @author  oliver de Cramer <oliverde8@gmail.com>
+ */
+interface WaypointDataListenerInterface
```
Comment (reaby) [inline src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/WaypointDataListenerInterface.php]: Why not ListenerInterfaceWaypointData as all other interfaces are prefixed, so they are easier to find at IDE

File: src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/LapDataListenerInterface.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace eXpansion\Framework\GameTrackmania\DataProviders\Listener;
+
+/**
+ * Class RaceDataListenerInterface
+ *
+ * @package eXpansion\Framework\GameTrackmania\DataProviders\Listener;
+ * @author  oliver de Cramer <oliverde8@gmail.com>
+ */
+interface LapDataListenerInterface
```
Comment (reaby) [inline src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/LapDataListenerInterface.php]: ListenerInterfaceLapData same here, it should be prefixed...

**Review by reaby — CHANGES_REQUESTED** [review]

File: src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/LapDataListenerInterface.php (hunk 2 above)
Comment (oliverde8) [reply src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/LapDataListenerInterface.php]: Right

File: src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/WaypointDataListenerInterface.php (hunk 1 above)
Comment (oliverde8) [reply src/eXpansion/Framework/GameTrackmania/DataProviders/Listener/WaypointDataListenerInterface.php]: Right

**Comment by oliverde8** [comment]
Renamed should be all ok now @reaby

**Review by reaby — APPROVED** [review]

**Pull request merged**